File: 01-Assignment-02-02-2026/Using_C/program_31.c
/* Program 31: Find Nth Fibonacci number using recursion */

// F(x) = F(x-1) + F(x-2) , Given F(0) = 0 & F(1) =1

#include <stdio.h>

int Fibonacci(int n) {
    if (n < 0)
        return -1;   
    if (n == 0)
        return 0;
    if (n == 1)
        return 1;

    return Fibonacci(n - 1) + Fibonacci(n - 2);
}

int main() {
    int n;
    printf("Enter a number: ");
    scanf("%d", &n);

    int result = Fibonacci(n);

    if (result == -1)
        printf("Fibonacci is not defined for negative numbers.");
    else
        printf("Fibonacci number of %d is %d.", n, result);

    return 0;
}

File: 01-Assignment-02-02-2026/Using_Java/Program31.java
/*
Program 31
Java program to find the Nth term in Fibonacci series using recursion
*/

import java.util.Scanner;

public class Program31 {

    public static int fibonacci(int n) {
        if (n == 0) {
            return 0;
        } else if (n == 1) {
            return 1;
        }
        return fibonacci(n - 1) + fibonacci(n - 2);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a number to get its Fibonacci value: ");
        int num = scanner.nextInt();

        if (num < 0) {
            System.out.println("Fibonacci is only defined for positive natural numbers.");
        } else {
            System.out.println("Fibonacci value of " + num + " is: " + fibonacci(num));
        }

        scanner.close();
    }
}
